contracts/src/PermissionedStakeTable.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { BN254 } from "bn254/BN254.sol";
+import { EdOnBN254 } from "./libraries/EdOnBn254.sol";
+
+/**
+ * @title SimpleStakeTable
+ * @dev An stake table mapping with owner-only access control.
+ */
+contract PermissionedStakeTable is Ownable {
+    event Added(NodeInfo[]);
+    event Removed(NodeInfo[]);
+
+    error StakerAlreadyExists(BN254.G2Point);
+    error StakerNotFound(BN254.G2Point);
+
+    struct NodeInfo {
```

Comment:
nit: Just thinking about naming, we've been using node and staker interchangeably? Could be good to just use one term to refer to this entity

Reply:
https://github.com/EspressoSystems/espresso-sequencer/issues/2384